### PR TITLE
Time series gapfilling improvement

### PR DESCRIPTION
**Pull request by Dreamlone**
The gap-filling module has been modified: it is now more stable. To confirm this, the tests have been redesigned. The following gaps configurations was tested:
image

As can be seen from the graph, the tests cover different boundary cases
The nan values are now automatically filled in

### CODE REVIEW

File: fedot/utilities/ts_gapfilling.py [hunk 1]
```diff
@@ -264,7 +268,7 @@ def forward_inverse_filling(self, input_data):
         return output_data
 
     @series_has_gaps_check
-    def forward_filling(self, input_data):
+    def forward_filling(self, input_data: Union[list, np.ndarray]):
```
Comment (nicl-nno) [inline fedot/utilities/ts_gapfilling.py:271]: А тут не должно быть List для совместимости с <3.9?

**Review by nicl-nno — APPROVED** [review]

File: fedot/utilities/ts_gapfilling.py (hunk 1 above)
Comment (Dreamlone) [reply fedot/utilities/ts_gapfilling.py]: Вроде я как раз сижу только на версиях FEDOT 3.7-3.8 и всё ок
[Поправил](https://github.com/nccr-itmo/FEDOT/pull/801/commits/7f8bf33c12b572632452696044fee0286d1cf7b3)
